src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -22,18 +47,24 @@ export const TodoFilter: React.FC = () => {
           type="text"
           className="input"
           placeholder="Search..."
+          value={query}
+          onChange={handleQueryChange}
         />
         <span className="icon is-left">
           <i className="fas fa-magnifying-glass" />
         </span>
 
         <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <button
-            data-cy="clearSearchButton"
-            type="button"
-            className="delete"
-          />
+          {!query.length ? '' : (
```

Comment:
```suggestion
          {!!query.length && (
```
FYI - if you don't need "else" case, i.e. want to render smth or not at all